shell/src/main/java/tachyon/shell/TFsShellUtils.java

```diff
@@ -74,4 +80,160 @@ public static String validatePath(String path, TachyonConf tachyonConf) throws I
+    File file = new File(inputPath);
+    if (!inputPath.contains("*")) {
+      List<File> res = new LinkedList<File>();
+      if (file.exists()) {
+        res.add(file);
+      }
+      return res;
+    } else {
+      String prefix = inputPath.substring(0, inputPath.indexOf(TachyonURI.WILDCARD) + 1);
+      String parent = new File(prefix).getParent();
+      return getFiles(inputPath, parent);
+    }
+  }
+  
+  /**
+   * The utility function used to implement getFiles
+   * It follows the same algorithm as getTachyonURIs
+   * @param inputPath The input file path (could contain wildcards)
+   * @param parent The directory in which we are searching matched files 
+   * @return A list of files that matches the input path in the parent directory
+   */
+  protected static List<File> getFiles(String inputPath, String parent) {
+    List<File> res = new LinkedList<File>();
+    File pFile = new File(parent);
+    if (!pFile.exists() || !pFile.isDirectory()) {
+      return res;
+    }
+    if (pFile.isDirectory() && pFile.canRead()) {
+      for (File file : pFile.listFiles()) {
+        if (match(file.getPath(), inputPath)) { // if it matches
+          res.add(file);
+        } else {
+          if (file.isDirectory()) { // if it is a folder, we do it recursively
+            TachyonURI dirURI = new TachyonURI(file.getPath());
+            String prefix = new TachyonURI(inputPath).getLeadingPath(dirURI.getDepth());
+            if (prefix != null && match(dirURI, new TachyonURI(prefix))) {
+              res.addAll(getFiles(inputPath, dirURI.getPath()));
+            }
+          }
+        }
+      }
+    }
+    return res;
+  }
+  
+  /**
+   * The characters that have special regex semantics
+   */
+  private static final Pattern SPECIAL_REGEX_CHARS = Pattern.compile("[{}()\\[\\].+*?^$\\\\|]");
+  
+  /**
+   * Escape the special characters in a given string
+   * @param str Input string
+   * @return the string with special characters escaped
+   */
+  private static String escape(String str) {
+    return SPECIAL_REGEX_CHARS.matcher(str).replaceAll("\\\\$0");
+  }
+  
+  /**
+   * Replace the wildcards with Java's regex semantics
+   */
+  private static String replaceWildcards(String text) {
+    return escape(text).replace("\\*", ".*");
+  }
+  
+  /**
+   * Return whether or not fileURI matches the patternURI
+   * @param fileURI The TachyonURI of a particular file
+   * @param patternURI The URI that can contain wildcards
+   * @return true if matches; false if not
+   */
+  protected static boolean match(TachyonURI fileURI, TachyonURI patternURI) {
+    return escape(fileURI.getPath()).matches(replaceWildcards(patternURI.getPath()));
+  }
+  
+  /**
+   * Return whether or not filePath matches patternPath
+   * @param filePath Path of a given file
+   * @param patternPath Path that can contain wildcards
+   * @return true if matches; false if not
+   */
+  protected static boolean match(String filePath, String patternPath) {
```

Comment:
@uronce-cc if I change this one to private, I cannot call it in the TfsShellUtilsTest :-(